packages/next/src/server/app-render/app-render.tsx

```diff
@@ -762,32 +758,61 @@ async function renderToHTMLOrFlightImpl(
     routeType: 'render',
   } as const
 
-  const onReactStreamRenderError = onInstrumentationRequestError
-    ? (err: Error) => onInstrumentationRequestError(err, req, errorContext)
-    : undefined
+  // Including RSC rendering and flight data rendering
+  function getRSCError(err: Error & { digest: string }) {
+    const digest = err.digest
+    if (!digestErrorsMap.has(digest)) {
+      digestErrorsMap.set(digest, err)
+    }
+    return err
+  }
+
+  function getSSRError(err: Error & { digest: string }) {
```

Comment:
non-blocking nit: I'd consider moving this to a shared type since we do it a lot